lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -855,6 +855,9 @@ def group_exists(client, module, vpc_id, group_id, name):
     if security_groups:
         groups = dict((group['GroupId'], group) for group in all_groups)
         groups.update(dict((group['GroupName'], group) for group in all_groups))
+        if vpc_id:
```

Comment:
The VPC is used to filter above - but it doesn't only return results matching the VPC, it returns matches that meet at least one of the filters.